aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1875,43 +1875,43 @@ Tensor _matmul_impl(
     // Why not t1->view(-1, sizes_1.back())?
     // If the last dim is 0, then view(-1, 0) won't work because the -1 becomes ambiguous.
     // This can happen in e.g. [3, 5, 0] @ [0, 0].
-    const auto sizes_1 = t1->sizes();
-    auto output_shape = DimVector(sizes_1.begin(), sizes_1.end() - 1);
+    const auto sizes_1 = t1->sym_sizes();
+    auto output_shape = SymDimVector(sizes_1.begin(), sizes_1.end() - 1);
```

Comment:
These changes shouldn't have been necessary; we have a python implementation of matmul and it should be triggered. Would be good to figure out why py dispatcher wasn't applied

Reply:
Random hunch: dynamo’s analysis pass that checks for input mutations before going to AOTAutograd needs to enable PyDispatcher. I can check later

Reply:
I looked into this and found that the `matmul` tests were failing for the matrix-vector case only. Making `at::outer` SymInt-aware fixes this, so I'm going to merge that instead of the changes here as we move to master.